test/gtest/ucs/test_datatype.cc

```diff
@@ -1356,3 +1359,358 @@ UCS_MT_TEST_F(test_datatype_ptr_map_safe, safe_put, num_threads)
+
+        for (auto &func : funcs) {
+            ucs_list_for_each(seg, &func.head, list) {
+                /* Test start, end and middle of the each segment */
+                points.insert(seg_start);
+                points.insert(seg->end);
+                points.insert(seg_start + (seg->end - seg_start) / 2);
+
+                seg_start = seg->end + 1;
+            }
+        }
+
+        for (size_t point : points) {
+            double expected = 0;
+            for (auto &func : funcs) {
+                expected += ucs_piecewise_func_apply(&func, point);
+            }
+
+            double actual = ucs_piecewise_func_apply(&result, point);
+
+            if (actual != expected) {
+                for (const auto &func : funcs) {
+                    print_piecewise_func(func);
+                }
+            }
+            ASSERT_EQ(actual, expected) << "point is " << point;
+        }
+    }
+
+    size_t get_random_value(size_t min, size_t max)
+    {
+        if (max == min) {
+            return min;
+        }
+        return min + (ucs::rand() % (max - min));
+    }
+
+    segment get_random_segment(size_t max_end = SIZE_MAX)
+    {
+        size_t start = get_random_value(0, max_end);
+        size_t end   = get_random_value(start, max_end);
+        auto func    = ucs_linear_func_make(get_random_value(0, 10),
+                                            get_random_value(0, 10));
+
+        return {start, end, func};
+    }
+
+    void piecewise_func_add_segment(ucs_piecewise_func_t *func, size_t start,
+                                    size_t end, ucs_linear_func_t trend)
+    {
+        ucs_status_t status;
+        status = ucs_piecewise_func_add_segment(func, start, end, trend);
+        ASSERT_UCS_OK(status);
+    }
+
+    void piecewise_func_add_inplace(ucs_piecewise_func_t *func1,
+                                    ucs_piecewise_func_t *func2)
+    {
+        ucs_status_t status;
+        status = ucs_piecewise_func_add_inplace(func1, func2);
+        ASSERT_UCS_OK(status);
+    }
+
+    void piecewise_func_fill_random(ucs_piecewise_func_t &func,
+                                    size_t max_custom_segments)
+    {
+        for (size_t s = 0; s < get_random_value(0, max_custom_segments); ++s) {
+            auto seg = get_random_segment(256);
+            piecewise_func_add_segment(&func, seg.start, seg.end, seg.func);
+        }
+    }
+
+    std::vector<size_t> get_segment_points(const segment &seg)
+    {
+        return {seg.start, seg.end, seg.start + (seg.end - seg.start) / 2};
+    }
+};
+
+UCS_TEST_F(test_piecewise_func, init) {
+    ucs_piecewise_func_t zero_func;
+    ucs_piecewise_func_init(&zero_func, 0, 0);
+    check_signle_segment_func(zero_func, UCS_LINEAR_FUNC_ZERO);
+
+    ucs_piecewise_func_t func;
+    ucs_piecewise_func_init(&func, 1, 1);
+    ucs_linear_func_t segment_func = ucs_linear_func_make(1, 1);
+    check_signle_segment_func(func, segment_func);
```

Comment:
typo

Reply:
This PR contains some outdated code state. This is done in this PR (https://github.com/openucx/ucx/pull/9569) which should be merged first.